docs/upgrade/2.0-3.0.md

```diff
@@ -0,0 +1,314 @@
+dependencies installed using various package managers (bower, npm, and
+composer). We realized this was an anti-pattern and have made some changes!
+
+First of all, we removed the use of bower as a dependency manager and have
+consolidated it's responsibilities into npm, lowering overall complexity. On top
+of that we have removed the need to commit packages and built versions of the
+front end assets.
+
+These things are now git ignored and will continue to be so. No more making
+sure to build and commit your assets before a deploy! It's recommended to build
+the client each time you deploy a new version of your code. This process is
+automated if you use the new Capistrano setup for deployment as well. (More on
+that later).
+
+### Customize your Madison instance!
+
+Before many of the changes in Madison 3.0, to customize your own version of
+Madison you had to maintain your own git fork of the codebase. If you wanted to
+change styles, language, page content, etc, it required that you make changes to
+files that were tracked in the git repository.
+
+We've made serious headway on that front! You can now customize your Madison
+instance using some git ignored files.
+
+* `client/app/sass/custom/` holds custom sass files in here to override existing
+  styles.
+* `client/app/locales/custom/` holds custom locale (language) files in here to
+  override language throughout the system.
+* Custom content pages and nav links can now be managed by site administrators
+  through the interface!
+
+These changes will allow you to keep tracking from the primary Madison repo and
+pulling in changes as we move forward without clobbering your instance-specific
+customizations.
+
+### Web server reconfiguration
+
+Because the server and client code are served from different directories, the
+apache / nginx configuration looks quite a bit different. To see an example of
+what this should look like now, check out the [example apache
+configuration](/docs/apache.conf.example) provided in our docs folder.
+
+The main thing is that we want the client code to be the primary document root,
+and have an alias set up that routes any requests prefixed with `/api` to the
+server (Laravel) code.
+
+You can also easily change whether you're looking at a development version of
+the client or a built production version by changing the document root to either
+`./client/app` for development or `./client/build` for production. This makes
+for easy confirmation of a successful build!
+
+### Document sponsorship moidel
+
+Instead of having a separate construct for "independent sponsors" vs "group
+sponsors", we simplified the model by merging the handling of "independent
+sponsors" into the groups model. Now when a user becomes and independent
+sponsor, they are being put into a specially marked group under the title of
+their own name. This simplifies handling of document sponsorship related
+permissions.
+
+### Summarize annotation model unification
+
+...
```

Comment:
Ah, yes. Need to fill this part in, that's an oversight on my part. I would like to consult with you on that though to make sure I get it nice and accurate. Let me know when you have time.